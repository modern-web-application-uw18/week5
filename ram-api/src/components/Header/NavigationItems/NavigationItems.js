import React from 'react';
import './NavigationItems.css';
import NavigationItem from './NavigationItem/NavigationItem';

const navigationItems = () => (
    <ul className="NavigationItems">
        <NavigationItem link="/" exact>Home</NavigationItem>
        <NavigationItem link="/favs">Favorites</NavigationItem>
        <NavigationItem link="/detail">Detail</NavigationItem>
    </ul>
)


export default navigationItems;
