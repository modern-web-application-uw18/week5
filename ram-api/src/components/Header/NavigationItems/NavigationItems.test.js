import React from 'react';
import { mount } from 'enzyme';
import NavigationItems from './NavigationItems'
import { MemoryRouter } from 'react-router-dom';
// import characters from '../../__mocks__/characters.json'

describe('<NavigationItems />', () => {
    it('Should render NavigationItems', () => {

        const wrapper = mount(
            <MemoryRouter>
                <NavigationItems />
            </MemoryRouter>
        );


        expect(wrapper).toMatchSnapshot();
    })
})




