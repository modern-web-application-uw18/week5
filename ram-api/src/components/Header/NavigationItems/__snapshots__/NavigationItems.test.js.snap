// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> Should render NavigationItems 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "lj6lhd",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "lj6lhd",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <navigationItems>
      <ul
        className="NavigationItems"
      >
        <navigationItem
          exact={true}
          link="/"
        >
          <li
            className="NavigationItem"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/"
            >
              <Route
                exact={true}
                path="/"
              >
                <Link
                  aria-current="true"
                  className="active"
                  replace={false}
                  style={Object {}}
                  to="/"
                >
                  <a
                    aria-current="true"
                    className="active"
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Home
                  </a>
                </Link>
              </Route>
            </NavLink>
          </li>
        </navigationItem>
        <navigationItem
          link="/detail"
        >
          <li
            className="NavigationItem"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/detail"
            >
              <Route
                path="/detail"
              >
                <Link
                  aria-current={false}
                  replace={false}
                  to="/detail"
                >
                  <a
                    aria-current={false}
                    href="/detail"
                    onClick={[Function]}
                  >
                    Detail
                  </a>
                </Link>
              </Route>
            </NavLink>
          </li>
        </navigationItem>
        <navigationItem
          link="/favs"
        >
          <li
            className="NavigationItem"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/favs"
            >
              <Route
                path="/favs"
              >
                <Link
                  aria-current={false}
                  replace={false}
                  to="/favs"
                >
                  <a
                    aria-current={false}
                    href="/favs"
                    onClick={[Function]}
                  >
                    Favorites
                  </a>
                </Link>
              </Route>
            </NavLink>
          </li>
        </navigationItem>
      </ul>
    </navigationItems>
  </Router>
</MemoryRouter>
`;
